internal/dag/conditions.go

```diff
@@ -30,13 +31,21 @@ func pathCondition(conds []projcontour.Condition) Condition {
 	}
 }
 
-func pathConditionsValid(conds []projcontour.Condition) bool {
+// pathConditionsValid validates a slice of Conditions.
+// It encodes the business rules about what is allowed for prefix Conditions.
+func pathConditionsValid(conds []projcontour.Condition, sw *ObjectStatusWriter, conditionsContext string) bool {
```

Comment:
I'm not a big fan of passing this conditionsContext string, but without it there is no way at all to tell which conditions block is invalid in a HTTPProxy document. Solving this properly requires solving #1575 as well, so this is here for now.